mitxgraders/helpers/calc/expressions.py

```diff
@@ -469,16 +367,16 @@ def parse_algebra(self):
                                |
                                ("." + number_part))
         # Combine() joints the matching parts together in a single token,
-        # and requires that the matching parts be contiguous
+        # and requires that the matching parts be contiguous (no spaces)
 
         # Define our suffixes
-        suffix = MatchFirst(Literal(k) for k in self.suffixes.keys())
+        suffix = Word(alphas + '%')
         suffix.setParseAction(self.suffix_parse_action)
 
-        # Construct a number as a group consisting of a text string (num) and an optional
-        # suffix num can include a decimal number and numerical exponent, and can be
+        # Construct number as a group consisting of a text string ("num") and an optional suffix.
+        # num can include a decimal number and numerical exponent, and can be
         # converted to a number using float()
-        # suffix is the suffix string that matches one of our suffixes
+        # suffix may contain alphas or %
         # Spaces are ignored inside numbers
         # Group wraps everything up into its own ParseResults object when parsing
         number = Group(
```

Comment:
Do we have a good reason to have `adjacent=False` here? I suspect I have a strong preference for True...

Reply:
Let's call it a bugfix. I'd like to train people not to write `1.5 e - 2` 😄